io_scene_niftools/modules/nif_import/property/texture/loader.py

```diff
@@ -109,62 +111,32 @@ def save_as_dds_override(self, stream):
                 elif channel.type == NifFormat.ChannelType.CHNL_ALPHA:
                     header.pixel_format.a_mask = mask
                 bit_pos += channel.bits_per_channel
-        header.caps_1.complex = 1
-        header.caps_1.texture = 1
-        header.caps_1.mipmap = 1
-        pixeldata.set_value(self.__get_pixeldata_stream())
-    elif self.pixel_format == NifFormat.PixelFormat.PX_FMT_DXT1:
-        # format used in Megami Tensei: Imagine and Bully SE
-        header.flags.caps = 1
-        header.flags.height = 1
-        header.flags.width = 1
-        header.flags.pixel_format = 1
-        header.flags.mipmap_count = 1
-        header.flags.linear_size = 0
-        header.height = self.mipmaps[0].height
-        header.width = self.mipmaps[0].width
-        header.linear_size = 0
-        header.mipmap_count = len(self.mipmaps)
-        header.pixel_format.flags.four_c_c = 1
-        header.pixel_format.four_c_c = DdsFormat.FourCC.DXT1
-        header.pixel_format.bit_count = 0
-        header.pixel_format.r_mask = 0
-        header.pixel_format.g_mask = 0
-        header.pixel_format.b_mask = 0
-        header.pixel_format.a_mask = 0
-        header.caps_1.complex = 1
-        header.caps_1.texture = 1
-        header.caps_1.mipmap = 1
-        pixeldata.set_value(self.__get_pixeldata_stream())
-    elif self.pixel_format in (NifFormat.PixelFormat.PX_FMT_DXT5,
+    elif self.pixel_format in (NifFormat.PixelFormat.PX_FMT_DXT1,
+                               NifFormat.PixelFormat.PX_FMT_DXT5,
                                NifFormat.PixelFormat.PX_FMT_DXT5_ALT):
-        # format used in Megami Tensei: Imagine
-        header.flags.caps = 1
-        header.flags.height = 1
-        header.flags.width = 1
-        header.flags.pixel_format = 1
-        header.flags.mipmap_count = 1
+        # format used in Megami Tensei: Imagine and Bully SE
+
         header.flags.linear_size = 0
-        header.height = self.mipmaps[0].height
-        header.width = self.mipmaps[0].width
         header.linear_size = 0
-        header.mipmap_count = len(self.mipmaps)
         header.pixel_format.flags.four_c_c = 1
-        header.pixel_format.four_c_c = DdsFormat.FourCC.DXT5
+        if self.pixel_format in (NifFormat.PixelFormat.PX_FMT_DXT1,):
+            header.pixel_format.four_c_c = DdsFormat.FourCC.DXT1
+        if self.pixel_format in (NifFormat.PixelFormat.PX_FMT_DXT5,
+                                 NifFormat.PixelFormat.PX_FMT_DXT5_ALT):
+            header.pixel_format.four_c_c = DdsFormat.FourCC.DXT5
         header.pixel_format.bit_count = 0
         header.pixel_format.r_mask = 0
         header.pixel_format.g_mask = 0
         header.pixel_format.b_mask = 0
         header.pixel_format.a_mask = 0
-        header.caps_1.complex = 1
-        header.caps_1.texture = 1
-        header.caps_1.mipmap = 1
-        pixeldata.set_value(self.__get_pixeldata_stream())
```

Comment:
Would this issue be caused by niftools/pyffi#69

Reply:
```
                if isinstance(self,
                              NifFormat.NiPersistentSrcTextureRendererData):
                    pixeldata.set_value(
                        ''.join(
                            ''.join([chr(x) for x in tex])
                            for tex in self.pixel_data))
                else:
                    pixeldata.set_value(''.join(self.pixel_data_matrix))
```

Reply:
Not really, it's an underlying code sanity check.